pkg/apis/velero/v1/snapshot.go

```diff
@@ -31,36 +25,45 @@ type SnapshotSpec struct {
 
 	// BackupUID is the UID of the Velero backup this snapshot
 	// is associated with.
+	// +optional
 	BackupUID string `json:"backupUID"`
 
 	// Location is the name of the VolumeSnapshotLocation where this snapshot is stored.
+	// +optional
 	Location string `json:"location"`
 
 	// PersistentVolumeName is the Kubernetes name for the volume.
-	PersistentVolumeName string `json:persistentVolumeName`
+	// +optional
+	PersistentVolumeName string `json:"persistentVolumeName"`
 
 	// ProviderVolumeID is the provider's ID for the volume.
+	// +optional
 	ProviderVolumeID string `json:"providerVolumeID"`
 
 	// VolumeType is the type of the disk/volume in the cloud provider
 	// API.
+	// +optional
 	VolumeType string `json:"volumeType"`
 
 	// VolumeAZ is the where the volume is provisioned
 	// in the cloud provider.
+	// +optional
 	VolumeAZ string `json:"volumeAZ,omitempty"`
 
 	// VolumeIOPS is the optional value of provisioned IOPS for the
 	// disk/volume in the cloud provider API.
+	// +optional
 	VolumeIOPS *int64 `json:"volumeIOPS,omitempty"`
 }
 
 type SnapshotStatus struct {
 	// ProviderSnapshotID is the ID of the snapshot taken in the cloud
 	// provider API of this volume.
+	// +optional
```

Comment:
I might be missing something, but why change this field to optional? Without it, we wouldn't be able to locate a volume snapshot.